src/include/rejson_api.h

```diff
@@ -20,7 +20,11 @@ typedef enum JSONType {
   JSONType_Object = 4,
   JSONType_Array = 5,
   JSONType_Null = 6,
-  JSONType__EOF
+  JSONType__EOF = 7, // keep this for backward compatibility
```

Comment:
What is this value means and where is it used?

Reply:
Used by RediSearch `JSONType`. Marks a non-existing type (maybe it was wrong to introduce it, but now it exists)

Reply:
Do we ever return such a value? How come we will have none existing type?

Reply:
We do not return it, but if we would use value `7` for a new type, an older RediSearch module might error.
Maybe it was used as the type for uninitialized value.